src/lib/ext-monitor.js

```diff
@@ -76,17 +80,49 @@ export default class ExtensionMonitor {
     startMonitor: () => this.startMonitor(),
     stopMonitor: () => this.stopMonitor(),
     sendAllLogs: () => ({ existingLogs: this.logs }),
+    setLoadedLogs: (detail) => this.setLoadedLogs(detail),
+    getLoadedLogs: (detail) => this.getLoadedLogs(detail),
+    setLoadedLogsTabId: (detail) => this.setLoadedLogsTabId(detail),
   };
 
+  setLoadedLogs({ fileName, logs }) {
+    let repeatTimes = 0;
+    for (const storedFileName of this.loadedLogs.keys()) {
+      if (storedFileName.includes(fileName)) {
+        repeatTimes++;
+      }
+    }
+
+    // if file exists with same name
+    if (repeatTimes) {
+      fileName = `${fileName}-${repeatTimes}`;
+    }
+
+    this.loadedLogs.set(fileName, logs);
+    return fileName;
+  }
+
+  getLoadedLogs({ fileName }) {
+    const logs = this.loadedLogs.get(fileName);
+    if (!logs) {
+      throw new Error(`The following log file is not found: ${fileName}`);
+    }
+    return logs;
+  }
+
+  setLoadedLogsTabId({ fileName, tabId }) {
+    this.loadedLogsTabIds.set(tabId, fileName);
+  }
+
   messageListener = (message) => {
-    const { requestType, requestTo } = message;
+    const { requestType, requestTo, detail } = message;
```

Comment:
Let's rename detail to something like `requestParams`, in the end it is what we pass to the requestType method and so it should be easier to understand what that property is.

Reply:
Thanks for the suggestion. Updated the name :-)